src/peft/peft_model.py

```diff
@@ -1845,6 +1846,31 @@ def __init__(
         super().__init__(model, peft_config, adapter_name, **kwargs)
         self.base_model_prepare_inputs_for_generation = self.base_model.prepare_inputs_for_generation
 
+        # Condition to check if embedding layer (`embed_tokens`) is added
+        # in `modules_to_save` and we want to ensure the `lm_head`
+        # does not diverge from the `embed_tokens` layer
+        if (
+            peft_config.task_type == "CAUSAL_LM"
```

Comment:
Why is it important?

Reply:
Yes, I'll make the changes such that we only worry about weight tying in case one of the tied layers is added in the modules to save